source/core/index.ts

```diff
@@ -930,12 +930,16 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 		if (defaults && !areHooksDefault) {
 			for (const event of knownHookEvents) {
 				const defaultHooks = defaults.hooks[event];
-
 				if (defaultHooks.length !== 0) {
+					let optionsHooks = options.hooks[event] as any[];
+					if (optionsHooks && optionsHooks.length !== 0) {
+						optionsHooks = optionsHooks.filter(hook => !(defaultHooks as any[]).includes(hook));
```

Comment:
This prevents intentional duplicates.

Reply:
Oh oopsies! I'll add some sort of flag instead of filtering through the hooks, unsure whether I should use `kIsNormalizedAlready` since it might have unintended side effects, we'll see after the tests run.